hal.cabal

```diff
@@ -131,7 +131,7 @@ test-suite hal-test
       ScopedTypeVariables
   ghc-options: -Wall -Wcompat -Wincomplete-record-updates -Wincomplete-uni-patterns -Wredundant-constraints -fno-warn-partial-type-signatures -fno-warn-name-shadowing -fwarn-tabs -fwarn-unused-imports -fwarn-missing-signatures -fwarn-incomplete-patterns -threaded -rtsopts -with-rtsopts=-N
   build-depends:
-      aeson >=1.2.0.0 && <1.6 || >=2.0.0.0 && <2.2.0.0
+      aeson >=1.2.0.0 && <1.6 || >=2.0.0.0 && <2.3.0.0
```

Comment:
I'd make this change in the `package.yaml` (which generates the cabal file) and move the bounds from the `test-suite` to the `library` section.